### PR TITLE
Titiler-cmr layer

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
**Related Ticket:** https://github.com/NASA-IMPACT/veda-ui/issues/1000

### Description of Changes

Application code changes - necessary changes
* renames ZarrPaintLayer to RasterPaintLayer, since it is not specific to Zarr
* adds a new type `titiler-cmr` and renames the `cmr` type to `cmr-stac` for disambiguation (although it is sure to be a bit confusing, since cmr-stac is for the use case of using the CMR STAC catalog and titiler-cmr is for the use case of tiling data in CMR but those requests are setup via collection records catalogued in VEDA STAC)
* adds a third hook `useTitilerCMR`. Because this third hook needs to return tiling parameters besides assetUrl, refactor the other hooks to also return a tileParams dictionary.

Application code - optional changes, simplifying some things
* Merges/removes duplication across MapLayer*TimeseriesProps to just one interface `MapLayerRasterTimeseriesProps` since the only thing that seemed to separate these types was `assetUrlReplacements`. This change is not strictly necessary for this functionality to work but it seemed to simplify things IMO.
* Simplify conditionals which set the datetime extent and periodicity. The VEDA STAC metadata for these CMR  collections uses `null` as the second datetime value in the temporal extent to indicate ongoing collections. This is the declared way to do so in the STAC collection specification, so it seems valid to me to allow for all non-vector collections to set the end datetime to the current date if it is null in the metadata.

Dataset changes
* Adds 2 datasets to demonstrate titiler-cmr is working for those datasets. We could go down to one since it seems cumbersome to manage multiple mock datasets for a single use case.
* Removes the GPM IMERG cmr-stac collection, since this overlaps with the titiler-cmr example and we can already cover the cmr-stac example with the TRMM dataset
* Adds the cmip6 kerchunk reference dataset, so there is a dataset to test with the "zarr" type

### Notes & Questions About Changes

I would be happy to go over these changes with one or more of the core UI developers.

### Validation / Testing

I tested these changes by opening up all of the mock datasets explore pages.

### CODE REVIEW

**Comment by abarciauskas-bgse** [comment]
Will address linting and ts-check later today

**Comment by abarciauskas-bgse** [comment]
Checks are passing but unsure about the changes in style-generators and why it seems like there is duplication of files there. Would love to pair with UI team member(s) to review!

**Comment by sandrahoang686** [comment]
@abarciauskas-bgse thanks for detailed description and for the explanation on #1000. I dont mind campfiring on this together for review! It will be more interactive though since I haven't worked with the style-generators much. I'm guessing the duplication of files may be related to this past PR https://github.com/NASA-IMPACT/veda-ui/pull/805 (specifically maybe this commit https://github.com/NASA-IMPACT/veda-ui/pull/805/commits/0b2dece29f2001707e231e6bccccee3509c207ef)

**Comment by sandrahoang686** [comment]
@abarciauskas-bgse the veda dashboard team is also working on deprecating the `/common/mapbox/` directory as we are are moving to consolidate our maps and mapping logic to a the new map provider and logic under dir `/common/map/`. This is why you may be seeing duplicated logic in both these places. Work is still happening on migrating current maps and logic to the new directory. I recommend if possible to try to minimize references to anything under `/common/mapbox/` and if the logic doesn't already exist under the new dir `/common/map/` if you would be comfortable moving this logic over, I can work with you to do that if as well

File: app/scripts/components/common/mapbox/layers/raster-paint-layer.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import { useEffect, useMemo } from 'react';
```
Comment (sandrahoang686) [inline app/scripts/components/common/mapbox/layers/raster-paint-layer.tsx:1]: 🔧 : We are working to deprecate this path. The new path for this logic should go under `/common/map/`. I believe you would want to migrate this over to `/common/map/style-generators/`.

File: app/scripts/components/common/map/style-generators/hooks.ts [hunk 2]
```diff
@@ -23,8 +23,13 @@ interface CMRResponseData {
   }[]
 }
 
-export function useZarr({ id, stacCol, stacApiEndpointToUse, date, onStatusChange }){
-  const [assetUrl, setAssetUrl] = useState('');
+interface STACforCMRResponseData {
+  collection_concept_id: string;
+  renders: Record<string, any>;
+}
```
Comment (sandrahoang686) [inline app/scripts/components/common/map/style-generators/hooks.ts:29]: I wonder 🤔 if the backend team has json schemas we could generate into typescript interfaces to import, interested.. I'll follow up with them on this

File: app/scripts/components/common/map/style-generators/hooks.ts [hunk 3]
```diff
@@ -90,11 +103,73 @@ export function useCMR({ id, stacCol, stacApiEndpointToUse, date, assetUrlReplac
...
+        const data: STACforCMRResponseData = await requestQuickCache({
+          url: `${stacApiEndpointToUse}/collections/${stacCol}`,
+          method: 'GET',
+          controller
+        });
+
+        let tileParams = {
+          concept_id: data.collection_concept_id,
+          datetime: date,
+          ...sourceParams
```
Comment (sandrahoang686) [inline app/scripts/components/common/map/style-generators/hooks.ts:152]: 🔧 : This matches the naming of local state variable, can we change the name so it isn't confusing.

File: app/scripts/components/common/map/style-generators/hooks.ts [hunk 4]
```diff
@@ -38,11 +43,19 @@ export function useZarr({ id, stacCol, stacApiEndpointToUse, date, onStatusChang
           controller
         });
 
-        setAssetUrl(data.assets.zarr.href);
+        const tileParams = {
```
Comment (sandrahoang686) [inline app/scripts/components/common/map/style-generators/hooks.ts:46]: 🔧 : This matches the local state variable's name, can we change this so it isn't confusing? Or i'd just set state directly with the object like so...
```
        if (data.assets.zarr.href) {
          setTileParams({
             url: data.assets.zarr.href,
             time_slice: date,
             ...sourceParams
           });
        }
```

**Comment by abarciauskas-bgse** [comment]
just a update that I am making some changes based on discussion with @sandrahoang686 (to use the new map component, to fix conflicts and to fix some variable naming). so will move this to draft until that is done and this is ready for review again.

**Comment by hanbyul-here** [comment]
Can you also provide details on how the temporal text/summaries should be handled? @abarciauskas-bgse   I remember we briefly talked about this before, but it's been a while. We are currently shimming the end date to the current day, and density to DAY if the data is missing the density & temporal extent's end date. Is this still the case?

**Comment by abarciauskas-bgse** [comment]
Closing to propose a new design for different layer types

**Pull request closed without merging**